core/src/client/mod.rs

```diff
@@ -485,14 +486,17 @@ pub enum IdKind {
 	String,
 	/// Number.
 	Number,
+	/// Custom generator.
+	Custom(IdGeneratorFn),
 }
 
 impl IdKind {
-	/// Generate an `Id` from number.
+	/// Generate an `Id` from number or from a registered generator.
 	pub fn into_id(self, id: u64) -> Id<'static> {
```

Comment:
Hey @niklasad1 
I tried to not refactor too much as a first approach.
For now the user can set the custom generator with `ClientBuilder::id_format` but I believe the `into_id` function is  unnecessarily receiving a new id as an argument in case the IdKind is custom.

In your opinion how would `into_id` function signature change?

Reply:
Do you think we can remove this method completely and replace `IdKind::Number` and `IdKind::String` with generators as well but that takes state/has state?

It could also be an iterator-like API where we just call next() on it.

Reply:
So the user when calling `ClientBuilder::id_format` should provide a generator or the default one for Number and String right?

Reply:
yes but we may need to change the name for it but I'm not sure.

Reply:
The only reason I dislike the Iterator pattern here is that we are forced to return an Option from next so also from next_id when there is no real need to return an option

Reply:
You don't need use the Iterator API, if you have something better in mind go ahead (it was just an example)

It's really up to you what you think is the best.